lib/chat_app_web/live/single_room_live.ex

```diff
@@ -91,4 +110,21 @@ defmodule ChatAppWeb.SingleRoomLive do
         end)
      )}
   end
+
+  def handle_info({:users_typing, username, status}, socket) do
+    case socket.assigns.username do
+      username -> {:noreply, socket}
+      _ ->
+        current_status = MapSet.member?(socket.assigns.users_typing, username)
+        animation_condition = current_status != status or current_status == status and current_status
+        users_typing = socket.assigns.users_typing
```

Comment:
This will always match the first clause.
`username` here is not the one from function parameters, it's just a symbolic name for any value - it could have been called any name, like `user_name`, `name`, `something`, `bob` etc.
Check out the [pin operator](https://elixir-lang.org/getting-started/pattern-matching.html#the-pin-operator).